syntax = "proto3";
import "google/protobuf/empty.proto";

service CoordinatorService {
    rpc StartJob(StartJobRequest) returns (StartJobReply) {}
    rpc LastJobStatus(google.protobuf.Empty) returns (LastJobStatusReply) {}
    rpc UploadFiles(UploadFilesRequest) returns (UploadFilesReply) {}
}

message StartJobRequest {
    string inputLocation = 1;
    string outputLocation = 2;
    string coordinatorCodeLocation = 3;
    string workerCodeLocation = 4;
}

message StartJobReply {
    string jobUuid = 1;
}

message LastJobStatusReply {
    message JobStatus {
        string jobUuid = 1;
        bool finished = 2;
        string currentStep = 3;
        int32 stepPartsFinished = 4;
        int32 stepPartsTotal = 5;
        string inputLocation = 6;
        string outputLocation = 7;
    }
    JobStatus status = 1;
}

message UploadFilesRequest {
    int32 numberOfFiles = 1;
    bool withoutDirectory = 2;
}

message UploadFilesReply {
    repeated string inputLocation = 1;
    repeated string uploadUrls = 2;
}
